specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,273 @@
+}
+
+/// This is the result for ExecuteScriptWithResult.
+[uuid(D2C59C5C-AD36-4CF4-87CF-2F5359F6D4CB), object, pointer_default(unique)]
+interface ICoreWebView2ExecuteScriptResult : IUnknown {
+
+  /// This property is true if ExecuteScriptWithResult successfully executed script with
+  /// no unhandled exceptions and the result is available in the ResultAsJson property
+  /// or via the TryGetResultAsString method.
+  /// If it is false then the script execution had an unhandled exception which you
+  /// can get via the Exception property.
+  [propget] HRESULT IsSuccess([out, retval] BOOL* value); 
+
+  /// If IsSuccess is true, then this property is the JSON representation of the result of the script execution
+  /// and otherwise returns E_INVALIDARG.
+  [propget] HRESULT ResultAsJson([out, retval] LPWSTR* jsonResult);
+
+  /// If IsSuccess is true and the result of script execution is a string, this method provides the value of the string result
+  /// and otherwise returns E_INVALIDARG.
+  HRESULT TryGetResultAsString([out, retval] LPWSTR* stringResult);
+
+  /// If IsSuccess is false, you can use this property to get the unhandled exception thrown by script execution
+  /// and otherwise returns E_INVALIDARG.
+  [propget] HRESULT Exception(
+      [out, retval] ICoreWebView2ExecuteScriptException** exception);
+}
+
+/// This is the callback for ExecuteScriptWithResult
+[uuid(CECDD25B-E6E8-4A4E-B890-BBF95932564F), object, pointer_default(unique)]
+interface ICoreWebView2ExecuteScriptWithResultCompletedHandler : IUnknown {
+
+  /// Provides the result of ExecuteScriptWithResult
+  HRESULT Invoke(
+      [in] HRESULT errorCode, 
+      [in] ICoreWebView2ExecuteScriptResult* result);
+}
+
+/// This is the interface for getting string and exception with ExecuteScriptWithResult
+[uuid(67E0B57B-1AC7-4395-9793-5E4EF9C4B7D9), object, pointer_default(unique)]
+interface ICoreWebView2_10 : ICoreWebView2_9 {
+  
+  /// Allows you to execute JavaScript and receive the successful result as JSON or as a string, or in the case of an unhandled exception you can get the exception.
+  /// if execution fails.
+  HRESULT ExecuteScriptWithResult(
+      [in] LPCWSTR javaScript,
+      [in] ICoreWebView2ExecuteScriptWithResultCompletedHandler* handler);
+}
+```
+
+## .NET and WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2;
+    runtimeclass CoreWebView2ExecuteScriptResult;
+    runtimeclass CoreWebView2ExecuteScriptException;
+
+    runtimeclass CoreWebView2
+    {
+        Windows.Foundation.IAsyncOperation<CoreWebView2ExecuteScriptResult>
+        ExecuteScriptWithResultAsync(String javaScript);
+    }
+
+    runtimeclass CoreWebView2ExecuteScriptResult
+    {
+        bool IsSuccess { get; };
+
+        String ResultAsJson { get; };
+
+        String TryGetResultAsString();
+        CoreWebView2ExecuteScriptException Exception { get; };
+    }
+
+    runtimeclass CoreWebView2ExecuteScriptException
```

Comment:
This should be updated from the changes to the COM. There's now a LineNumber, ColumnNumber, and the name of the runtimeclass should now be CoreWebView2ScriptException. If you run build-apiwriter.ps1 you can copy the new results out of the generated Microsoft.Web.WebView2.Core.IDL file.